src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -20,9 +20,9 @@ namespace Microsoft.CodeAnalysis.CodeActions
     public abstract class CodeAction
     {
         /// <summary>
-        /// A short title describing the action that may appear in a menu.
+        /// A short localizable title describing the action that may appear in a menu.
         /// </summary>
-        public abstract string Title { get; }
+        public abstract LocalizableString Title { get; }
```

Comment:
This means that all the places where we access CodeAction.Title and invoke ToString or Equals Or GetHashCode on it can throw, and unless guarded with an exception handler.
Problem is simpler for LocalizableString fields of DiagnosticDescriptor as it is a sealed type and we make all the localizable strings Exception safe in the constructor of DiagnosticDescriptor itself. Given that CodeAction is an abstract type, and this property is also a public abstract, we can't do that here.

We need to do couple of things at all places that access CodeAction.Title in IDE layer:
1. Handle exceptions from it explicitly with a try catch OR invoke MakeExceptionSafe on LocalizableString.
2. Report diagnostics for exceptions from this LocalizableString.

Doing this at all current call sites is feasible, but likely going to be a maintenance nightmare.